chromium_src/third_party/blink/renderer/modules/mediastream/media_devices.cc

```diff
@@ -33,7 +33,10 @@ void FarbleMediaDevices(ExecutionContext* context,
     // domain+session key.
     std::mt19937_64 prng =
         BraveSessionCache::From(*context).MakePseudoRandomGenerator();
-    std::shuffle(media_devices->begin(), media_devices->end(), prng);
+    MediaDeviceInfoVector::iterator it_begin = media_devices->begin();
+    MediaDeviceInfoVector::iterator it_end = media_devices->end();
+    if (it_begin != it_end) ++it_begin;
+    std::shuffle(it_begin, it_end, prng);
```

Comment:
I think we can do all of this only when `media_devices.size() > 2`, i.e. we can do an early return at the very start and then access `settings` and do all that shuffle only if we have to shuffle anything.

Reply:
Done in https://github.com/brave/brave-core/commit/8e6154354503320256bc9581169c94a537f73149